step-templates/apollo-rover-check-subgraph-schema.json

```diff
@@ -0,0 +1,91 @@
+{
+  "Id": "a0f2d754-9cae-488d-820c-4798985fb8d9",
+  "Name": "Apollo Rover - Check Subgraph Schema",
+  "Description": "Check subgraph schema against Apollo Studio. This should be run before deploying the subgraph service when promoting to higher environments.\n\nThe script installs the [Apollo Rover CLI](https://apollographql.com/docs/rover) and runs the `rover subgraph check` command.",
+  "ActionType": "Octopus.Script",
+  "Version": 1,
+  "CommunityActionTemplateId": null,
+  "Packages": [
+    {
+      "Name": "Schema Package",
+      "PackageId": null,
+      "FeedId": "feeds-builtin",
+      "AcquisitionLocation": "Server",
+      "Properties": {
+        "Extract": "True",
+        "SelectionMode": "deferred",
+        "PackageParameterName": "ApolloSchemaPackage",
+        "Purpose": ""
+      }
+    }
+  ],
+  "Properties": {
+    "Octopus.Action.Script.ScriptSource": "Inline",
+    "Octopus.Action.Script.Syntax": "Bash",
+    "Octopus.Action.Script.ScriptBody": "ROVER_VERSION=$(get_octopusvariable \"ApolloRoverVersion\")\n[ -z \"$ROVER_VERSION\" ] || [ \"$ROVER_VERSION\" == \"latest\" ] && ROVER_VERSION=\"latest\" || ROVER_VERSION=\"v$ROVER_VERSION\"\n\ncurl -sSL https://rover.apollo.dev/nix/$ROVER_VERSION | sh\n\nexport APOLLO_KEY=$(get_octopusvariable \"ApolloKey\")\n\nrover subgraph check $(get_octopusvariable \"ApolloGraphRef\") \\\n  --name $(get_octopusvariable \"ApolloSubgraphName\") \\\n  --schema $(get_octopusvariable \"ApolloSchema\")\n"
```

Comment:
I tried running this script, and got the following output:

```
Extracting package '/home/Octopus/Files/catalogue@S0.0.8@347546F36BD72C489F116EC29A9E1BC9.zip' to '/home/Octopus/Work/ServerTasks-603503-cf13ccbc-cb5c-4fce-a1b8-929d0f229d44/Schema Package' 
downloading rover from https://github.com/apollographql/rover/releases/download/v0.21.0/rover-v0.21.0-x86_64-unknown-linux-gnu.tar.gz 
writing binary to /home/tentacle/.rover/bin/rover 
rover was successfully installed. Great! 
To get started you need Rover's bin directory (/home/tentacle/.rover/bin) in your PATH environment variable. Next time you log in this will be done automatically. 
To configure your current shell, you can run: 
exec /bin/bash -l 
If you would like to disable Rover's anonymized usage collection, you can set APOLLO_TELEMETRY_DISABLED=1 
You can check out our documentation at https://go.apollo.dev/r/docs. 
/home/Octopus/Work/ServerTasks-603503-cf13ccbc-cb5c-4fce-a1b8-929d0f229d44/Script.sh: line 8: rover: command not found 
Fatal
The remote script failed with exit code 127 
Fatal
The action Apollo Rover - Publish Subgraph Schema on a Worker failed 
```

The call to the `rover` executable should be an absolute path like `~/.rover/bin/rover` to account for the fact that the `PATH` environment variable will not be updated just after the tool is installed.

Reply:
I'd also recommend redirecting the output of most commands from stderr to stdout. Octopus treats stderr as an error, which is not technically correct (stderr is also used for informational messages), but is how Octopus works. Otherwise the entire script appears to produce errors:

![image](https://github.com/OctopusDeploy/Library/assets/160104/a64165d6-67e7-4f05-b1ce-cccc9c6c0a8b)

Reply:
> I'd also recommend redirecting the output of most commands from stderr to stdout. Octopus treats stderr as an error, which is not technically correct (stderr is also used for informational messages), but is how Octopus works.

That is good to know. I'll redirect `stderr` and call `rover` from its installed dir.